src/components/Filters/components/Input/Input.tsx

```diff
@@ -44,25 +49,37 @@ export const StyledInput = styled.input`
   :-ms-input-placeholder {
     color: ${getFormStyle('placeholderColor')};
   }
+
+  ${({ isInvalid }) =>
+    isInvalid &&
+    css`
+      ${stateStyles}
+      border-color: ${getFormStyle('invalidBorderColor')};
+    `}
 `;
 
 const Input: React.FC<InputProps> = ({
   value = '',
   onChange,
   maxLength,
-  pattern,
-  patternMessage,
+  pattern = patterns.string.pattern,
+  patternMessage = patterns.string.patternMessage,
+  validate = validateDefault,
   placeholder = 'String',
+  isDisabled = false,
 }) => {
-  const handleOnChange = (event) => {
-    if (patternMessage) {
-      const { target } = event;
-      target.setCustomValidity(
-        target.validity.patternMismatch ? patternMessage : '',
-      );
-    }
+  const [isInvalid, setIsInvalid] = useState(false);
 
+  const handleOnValidate = (event) => {
+    const invalid = onValidate(event, validate, patternMessage);
+    setIsInvalid(invalid);
+  };
+
+  const handleOnChange = (event) => {
     onChange(event);
+    if (isInvalid) {
```

Comment:
Not sure it's a good idea to call validate function after the actual change. I'd say the onChange shouldn't happen when the field is invalid IMO

Reply:
`onChange` will change Filters state by wrong value and then Filters state is exposed by another public prop `onChange`, it shouldn't probably happen. I would move it into else after validation check. `Input` is controlled component so if `event.target` will be invalid then `onChange` will not be called and user will not be able to continue to input wrong values. I would say it is fine

Reply:
When I tried this, the field value could no longer be updated when invalid. It's because onChange is tied to `handleValueChange` in `FilterRow`. Any suggestions @PrincAm ?

Reply:
Not even by backspace? If not we have to leave it because we control the state as I wrote

Reply:
There could be also issue with copy pasting the value, we have to bear it in mind